### PR TITLE
Add filtering to poll_accept

### PR DESCRIPTION
**Pull request by uglyoldbob**
The existing solution for filtering did not compile for me, this one works well.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -340,7 +340,17 @@ mod hyper_impl {
             self: Pin<&mut Self>,
             cx: &mut Context<'_>,
         ) -> Poll<Option<Result<Self::Conn, Self::Error>>> {
-            self.poll_next(cx)
+            match self.poll_next(cx) {
+                Poll::Ready(Some(res)) => match res {
+                    Ok(val) => Poll::Ready(Some(Ok(val))),
+                    Err(e) => {
+                        println!("error {}", e);
+                        Poll::Pending
```
Comment (tmccombs) [inline src/lib.rs:348]: This should be `Poll::Ready(Some(Err(...)))` not pending

File: src/lib.rs [hunk 2]
```diff
@@ -340,7 +340,17 @@ mod hyper_impl {
             self: Pin<&mut Self>,
             cx: &mut Context<'_>,
         ) -> Poll<Option<Result<Self::Conn, Self::Error>>> {
-            self.poll_next(cx)
+            match self.poll_next(cx) {
+                Poll::Ready(Some(res)) => match res {
+                    Ok(val) => Poll::Ready(Some(Ok(val))),
+                    Err(e) => {
+                        println!("error {}", e);
```
Comment (tmccombs) [inline src/lib.rs:347]: We probably shouldn't have a println here

**Review by tmccombs — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (uglyoldbob) [reply src/lib.rs]: That makes sense. I'll remove that.

File: src/lib.rs (hunk 1 above)
Comment (uglyoldbob) [reply src/lib.rs:347]: I still encountered issues with the hyper server stopping when I made that change, but returning Poll::Pending did not have that problem.

**Comment by tmccombs** [comment]
Perhaps you could give me more context on what you are trying to accomplish here? The code in https://github.com/tmccombs/tls-listener/blob/main/examples/http-stream.rs#L25 compiles, and runs fine for me. 

The change you have isn't acceptable as is, because it changes the behavior so that instead of allowing the consumer of the library to handle TLS errors in whatever manner is appropriate for that application, it always simply ignores the error. Applications may wish to do something else in this scenario. For example, log it using a logging framework, print the error to stderr, send it directly to syslog or some other logging system, downgrade to treating it as non-tls traffic, or even add the client ip to a blocklist if there are multiple cases from the same client, etc.

I'd be happy to help you figure out what you have isn't working if you provide more information, such some sample source code (if different than the example I gave), what version of rust you are compiling with, what version of this library you are using, etc.

**Comment by uglyoldbob** [comment]
That all makes sense. I missed a function call to hyper::server::accept in the http stream example.

**Pull request closed without merging**